ch_tools/chadmin/internal/zookeeper.py

```diff
@@ -386,38 +387,97 @@ def get_host_mention_in_task(zk, host, ddl_task_path):
         try:
             for line in zk.get(ddl_task_path)[0].decode().strip().splitlines():
                 if "hosts: " in line:
-                    escaped_host = escape_for_zookeeper(host)
-                    pos = line.find(escaped_host)
-                    # Not found
-                    if pos == -1:
-                        return None
-
-                    return line[pos : line.find("'", pos)]
+                    host_names_with_ports = ast.literal_eval(line[len("hosts: ") :])
+                    result = {
+                        hosts_with_port[: hosts_with_port.find(":")]: hosts_with_port
+                        for hosts_with_port in host_names_with_ports
+                    }
+                    return result
         except NoNodeError:
             pass
 
+    def _check_ddl_task_finished(zk, ddl_number_of_hosts, ddl_path):
+        """
+        Check that ddl is finished.
+        """
+        if not zk.exists(ddl_path):
+            return True
+
+        return ddl_number_of_hosts == len(
+            _get_children(zk, os.path.join(ddl_path, "finished/"))
+        )
+
+    def _ensure_ddl_tasks_finished(zk, dll_tasks_to_remove):
+        """
+        Go thought list of tasks to remove and tries to wait until all of them finish.
+        """
+        wait_time = 0.3
+        max_attemps = 20
+        all_tasks_finished: bool
+        for _ in range(0, max_attemps):
+            all_tasks_finished = True
+            for ddl_data in dll_tasks_to_remove:
+                ddl_path = ddl_data[0]
+                ddl_size = ddl_data[1]
+                if not _check_ddl_task_finished(zk, ddl_size, ddl_path):
+                    all_tasks_finished = False
+                    break
+
+            if all_tasks_finished:
```

Comment:
We can [use](https://docs.python.org/3/tutorial/controlflow.html#break-and-continue-statements-and-else-clauses-on-loops) `else:` after `for` loops instead of a flag.

Reply:
I meant that we can get rid of `all_tasks_finished` at all, doing early return